### PR TITLE
Run 'grafiti' as cleanup step after AWS smoke test failure

### PR DESCRIPTION
**Pull request by estroz**
Jenkinsfile: removed grafiti docker container dependency
tests/smoke/aws: removed grafiti docker container dependency

Environment variables required by `grafiti` to delete orphaned cluster resources are present within `tectonic-builder` containers and not accessible by a `grafiti` container. Cluster cleanup should be done by the same container that created the cluster.

### CODE REVIEW

**Comment by Quentin-M** [comment]
cool

**Comment by kans** [comment]
| Cluster cleanup should be done by the same container that created the cluster.
👍

`go get github.com/coreos/grafiti/cmd/grafiti`

This makes me a bit nervous because if grafiti ever breaks, we probably won't notice until we hit limits.

**Comment by estroz** [comment]
@kans any suggestions on how to lock the version (without using Docker image tags)?

**Comment by estroz** [comment]
ok to test

**Comment by estroz** [comment]
ok to test

**Comment by estroz** [comment]
ok to test

**Comment by estroz** [comment]
ok to test

**Comment by estroz** [comment]
ok to test

**Comment by mxinden** [comment]
In regards to extending the Tectonic builder Dockerfile I would like to start the discussion whether we should break this container up into smaller containers. 

Right now the image:
* is 600 MB big, 
* takes quite a while to  build, 
* and needs to be updated in many places each time we update one small build step.

If I am not mistaking we could run the Grafiti steps inside a small vanilla golang image with just Grafiti installed inside.

Let me know what you think. I do see the advantages of a single image that we can use everywhere and I am not sure what the best practice should be here.

**Comment by Quentin-M** [comment]
The release automation work that I am starting soon will definitely include a revamp of the building process in order to make it streamlined and producing a release tarball everytime.

**Comment by estroz** [comment]
@mxinden I agree. The current images' size and build time are too large. For now this is the best solution though, until @Quentin-M revamps the release process. There is too much state necessary to run grafiti contained in the smoke scripts to run grafiti in its own container at this point.

**Comment by mxinden** [comment]
@estroz @Quentin-M This plan sounds good to me. Let's split the Tectonic builder image up, once we rework the release process. Thanks.

File: Jenkinsfile [hunk 1]
```diff
@@ -217,37 +238,44 @@ pipeline {
...
+                      """
                     }
                   }
                 }
               }
             }
-          },
-          "SmokeTest: Azure": {
+          }
+/*          "SmokeTest: Azure": {
```
Comment (estroz) [inline Jenkinsfile]: Changed the title back. All this is commented out to test failure recovery, and will be un-commented before merge.

**Comment by estroz** [comment]
Update: `grafiti delete` is not picking up `AWS_REGION` or the `region` config field when run via `smoke.sh`. I'm in the process of debugging why it isn't.

**Comment by estroz** [comment]
ok to test

**Comment by estroz** [comment]
ok to test

**Comment by estroz** [comment]
ok to test

**Comment by estroz** [comment]
ok to test

**Comment by estroz** [comment]
@mxinden @kans @Quentin-M good to review

**Comment by mxinden** [comment]
@estroz Could you rename the PR. If I understand correctly #1442 adds *Grafiti* to the Tectonic builder Docker image. This PR adds *Grafiti* to our Jenkins test pipeline.

**Pull request merged**